core/src/main/java/org/apache/iceberg/UpdateSnapshotReferencesOperation.java

```diff
@@ -85,16 +86,30 @@ public UpdateSnapshotReferencesOperation removeTag(String name) {
   }
 
   public UpdateSnapshotReferencesOperation renameBranch(String name, String newName) {
-    Preconditions.checkNotNull(name, "Branch to rename cannot be null");
-    Preconditions.checkNotNull(newName, "New branch name cannot be null");
+    renameRef(name, newName, SnapshotRefType.BRANCH);
+    return this;
+  }
+
+  public UpdateSnapshotReferencesOperation renameTag(String name, String newName) {
+    renameRef(name, newName, SnapshotRefType.TAG);
+    return this;
+  }
+
+  private void renameRef(String name, String newName, SnapshotRefType type) {
     Preconditions.checkArgument(!name.equals(SnapshotRef.MAIN_BRANCH), "Cannot rename main branch");
+    String refTypeInMessage = type == SnapshotRefType.BRANCH ? "Branch" : "Tag";
```

Comment:
nit: could just reference the enum value as a string in the checks, could make the argument about having more than two types of snapshots in the future, but i wouldn't know what those could be. this does help with readability though